api-authorization/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,36 @@
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const auth = req.headers.authorization;
+  // const auth = req.get('Authorization');   ---Get would be useful for content-type (includes invalid character)---
+  const token = auth?.split('Bearer ')[1];
+  if (!req.headers || !token) {
```

Comment:
If `req.headers` is undefined you will get an error on line 10, so the check here isn't very useful.

Usually in Express we get the headers with `req.get()`.